students/ian_letourneau/mailroom/mailroomOO.py

```diff
@@ -0,0 +1,236 @@
+#!/usr/bin/env python3
+# Ian Letourneau
+# 5/11/2018
+# A script that stores values for donors and their donations.
+# The script can process requests from the user on various functions
+# to utilize these values
+import datetime
+
+# Build a dictionary of existing donors
+
+
+class Donor:
+    """An object that contains information for donors and their donations.
+        attribute name: donors name.
+        attribute donation_list: a list of given donations."""
+
+    def __init__(self, name, donations=None):
+        """Initialization for Donor objects.
+
+        Params name: Will be the name given to Donor object
+        Params donations: a value or list of values to be 
+            combined into donation_list attribute."""
+        self.name = name
+        if isinstance(donations, list):
+            self.donation_list = donations
+        elif donations == None:
+            self.donation_list = []
+        else:
+            self.donation_list = [donations]
+
+    @property
+    def total(self):
+        """Getter for donation total."""
+        return sum(self.donation_list)
+
+    @property
+    def name_length(self):
+        """Getter for donor name length."""
+        return len(self.name)
+
+    @property
+    def don_num(self):
+        """Getter for number of donations."""
+        return len(self.donation_list)
+
+    @property
+    def average(self):
+        """Getter for average donation amount."""
+        return (self.total/len(self.donation_list))
+
+    def append(self, donation):
+        """Append function to tack on future donations."""
+        self.donation_list.append(donation)
+
+    def __eq__(self, other):
+        """Override equals comparison method."""
+        return self.total == other.total
+
+    def __lt__(self, other):
+        """Override less than comparison method."""
+        return self.total < other.total
+
+    def send_thank_you(self, donation):
+        """A function to send thank you letter utilizing given amount and
+        referenced donor object."""
+
+        # Append the amount to the running donation_list.
+        self.append(donation)
```

Comment:
it is probably best for this function to be `get_thank_you_letter` and keep append logic outside of this as it will feels a little bit like a side-effect.